sycl/doc/extensions/experimental/sycl_ext_oneapi_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -419,106 +551,30 @@ struct tpu_params<tpu::amx, Ta, Tb, Tc, 0, 0, 0,
   template <typename Group, layout LayoutB>
   using joint_matrix_b = joint_matrix<Group, Tb, use::b, K, N, LayoutB>;
   template <typename Group>
-  using joint_matrix_accumulator = joint_matrix<Group, Tc, use::accumulator, M, N>;
-
-  static constexpr uint32_t numtiles = 8;
-  static constexpr scope_t scopes[] = {scope_t::sub_group};
-  static constexpr int num_scopes = sizeof(scopes) / sizeof(scope_t);
-  struct combination {
-    uint32_t max_msize;
-    uint32_t max_nsize;
-    uint32_t max_ksize;
-    uint32_t msize;
-    uint32_t nsize;
-    uint32_t ksize;
-    matrix_type atype;
-    matrix_type btype;
-    matrix_type accumulatortype;
-  };
-  // In this case, the combinations array contain only the combinations that correspond to the Ta, Tb, and Tc 
-  // types that the user provided
-  static constexpr combination combinations[] = {
-      {16, 16, (sizeof(Ta) == 1) ? 64 : 32}};
-  static constexpr int num_combinations =
-      sizeof(combinations) / sizeof(combination);
-};
-
-// General query form:
-// types are not given, no default sizes and no implicit matrix construction
-template <int sM, int sN, int sK>
-struct tpu_params<tpu::amx, void, void, void, sM, sN, sK> {
-  static constexpr uint32_t numtiles = 8;
-  static constexpr scope_t scopes[] = {scope_t::sub_group};
-  static constexpr int num_scopes = sizeof(scopes) / sizeof(scope_t);
-  struct combination {
-    uint32_t max_msize;
-    uint32_t max_nsize;
-    uint32_t max_ksize;
-    uint32_t msize;
-    uint32_t nsize;
-    uint32_t ksize;
-    matrix_type atype;
-    matrix_type btype;
-    matrix_type accumulatortype;
-  };
-  
-  static constexpr combination combinations[] = {
-      {16, 16, 64, 0, 0, 0, matrix_type::sint8, matrix_type::sint8, matrix_type::sint32},
-      {16, 16, 64, 0, 0, 0, matrix_type::sint8, matrix_type::uint8, matrix_type::sint32},
-      {16, 16, 64, 0, 0, 0, matrix_type::uint8, matrix_type::sint8, matrix_type::sint32},
-      {16, 16, 64, 0, 0, 0, matrix_type::uint8, matrix_type::uint8, matrix_type::sint32},
-      {16, 16, 32, 0, 0,0, matrix_type::bf16, matrix_type::bf16, matrix_type::fp32}};
-  static constexpr int num_combinations =
-      sizeof(combinations) / sizeof(combination);
+  using joint_matrix_accumulator = joint_matrix<Group, Tc,
+  use::accumulator, M, N>;
 };
```

Comment:
The synopses above have too many implementation details.  I suggest defining it something like:

```
// This is the validation form, when all template parameters are specified.
template<tpu Dev, typename Ta, typename Tb, typename Taccumulator, int sM=0, int sN=0, int sK=0>
struct matrix_params {
  // An implementation typically uses static_assert here to trigger a compilation error when
  // the matrix types or shapes are not supported by the device identified by "Dev".

  using type_a = /* implementation defined */;
  using type_b = /* implementation defined */;
  using type_accumulator = /* implementation defined */;

  static constexpr size_t M = sM;
  static constexpr size_t N = sN;
  static constexpr size_t K = sK;

  template <typename Group, layout Layout>
  using joint_matrix_a = joint_matrix<Group, Ta, use::a, sM, sK, Layout>;

  template <typename Group, layout Layout>
  using joint_matrix_b = joint_matrix<Group, Tb, use::b, sK, sN, Layout>;

  template <typename Group>
  using joint_matrix_accumulator = joint_matrix<Group, Taccumulator, use::accumulator, sM, sN>;
};

// This is the default values form, where the matrix dimensions are omitted.
template<tpu Dev, typename Ta, typename Tb, typename Taccumulator>
struct matrix_params<Dev, Ta, Tb, Taccumulator, 0, 0, 0> {
  // An implementation typically uses static_assert here to trigger a compilation error when
  // the matrix types are not supported by the device identified by "Dev".

  using type_a = /* implementation defined */;
  using type_b = /* implementation defined */;
  using type_accumulator = /* implementation defined */;

  static constexpr size_t M = /* implementation defined */;
  static constexpr size_t N = /* implementation defined */;
  static constexpr size_t K = /* implementation defined */;

  template <typename Group, layout Layout>
  using joint_matrix_a = /* implementation defined */;

  template <typename Group, layout Layout>
  using joint_matrix_b = /* implementation defined */;

  template <typename Group>
  using joint_matrix_accumulator = /* implementation defined */;
};
```

Actually, I'm not sure about this part:

```
  using type_a = /* implementation defined */;
  using type_b = /* implementation defined */;
  using type_accumulator = /* implementation defined */;
```

It depends on how you answer my question above about the meaning of these type aliases.

Reply:
Since:
using type_a = Ta;
using type_b = Tb;
using type_accumulator = Taccumulator;

I am wondering why we needed to add these alias in the first place.

Reply:
OK, let's remove then for now.  We can add them back later if they seem useful.